### PR TITLE
Added unique pragma

### PR DESCRIPTION
**Pull request by Clonkk**
https://github.com/moigagoo/norm/issues/87
* Added unique pragma

### CODE REVIEW

File: src/norm/pragmas.nim [hunk 1]
```diff
@@ -16,3 +16,5 @@ template ro* {.pragma.}
 
   ``id`` field is ``ro`` by default.
   ]##
+
+template unique* {.pragma.}
```
Comment (moigagoo) [inline src/norm/pragmas.nim:20]: @Clonkk this needs a docstring.

**Review by moigagoo — CHANGES_REQUESTED** [review]
@Clonkk this looks pretty obvious but it's hard to see how that pragma affects the way queries are generated and what's the profit of using it for the user. This should be addressed in tests.

Also, please add a changelog entry for this change.

**Review by moigagoo — CHANGES_REQUESTED** [review]
Please add a test and resolve the conflict in the changelog.

File: changelog.rst [hunk 2]
```diff
@@ -96,8 +96,11 @@ The new algorithm adds alias for each joined table. The alias is named after the
 
 -   [r] Update Nim version to 1.4.0 in Dockerfile.
 
+-   [+] Added a define to enable logging
```
Comment (moigagoo) [inline changelog.rst:99]: This line is redundant, see the line below.

File: src/norm/pragmas.nim [hunk 3]
```diff
@@ -16,3 +16,7 @@ template ro* {.pragma.}
...
+##[ Mark field  as unique
```
Comment (moigagoo) [inline src/norm/pragmas.nim:21]: This should be a single line comment.

File: tests/tpostgresunique.nim [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import models
+
+
+const
+  dbHost = "postgres"
+  dbUser = "postgres"
+  dbPassword = "postgres"
+  dbDatabase = "postgres"
+
+proc duplicateInsert(dbConn: DbConn)
```
Comment (moigagoo) [inline tests/tpostgresunique.nim:15]: What's the purpose of this forward declaration?

File: tests/tpostgresunique.nim [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  teardown:
+    close dbConn
+    resetDb()
+
+  test "Duplicate insert":
+    # Check test fail with DbError
+    expect DbError:
+      duplicateInsert(dbConn)
+
+proc duplicateInsert(dbConn: DbConn)=
```
Comment (moigagoo) [inline tests/tpostgresunique.nim:37]: I think this is way overcomplicated. You could just have all that as part of the test, inside the test body.

File: tests/tpostgresunique.nim [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  block:
+    let
+      toy = newToy(123.45)
+      pet = newPet("cat", toy)
+
+    var person = newPerson("Alice", pet)
+    dbConn.insert(person)
+    check true
```
Comment (moigagoo) [inline tests/tpostgresunique.nim:45]: I'm not really happy with those check true and check false things. Explicit, meaningful checks are better.

File: tests/tpostgresunique.nim [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+  block:
+    let
+      toy = newToy(36.66)
+      pet = newPet("dog", toy)
+
+    var person = newPerson("Alice", pet)
+    dbConn.insert(person)
+    # Check that test does not reach this point
```
Comment (moigagoo) [inline tests/tpostgresunique.nim:54]: If the whole proc were just part of the test, there would be no need in this trick.

File: tests/tpostgresunique.nim [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  setup:
+    resetDb()
+    let dbConn = open(dbHost, dbUser, dbPassword, dbDatabase)
+
+  teardown:
+    close dbConn
+    resetDb()
+
+  test "Duplicate insert":
+    # Check test fail with DbError
```
Comment (moigagoo) [inline tests/tpostgresunique.nim:33]: Unnecessary comment, it's pretty obvious what the check is from the next line.

File: tests/tsqliteunique.nim [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import options
+import os
+
+import norm/[model, sqlite]
+
+import models
+
+const dbFile = "test.db"
+
+proc duplicateInsert(dbConn: DbConn)
```
Comment (moigagoo) [inline tests/tsqliteunique.nim:12]: See comments for Postgres.

**Review by moigagoo — CHANGES_REQUESTED** [review]

**Comment by Clonkk** [comment]
> What's the point of this forward declaration?

It's mostly habits from developing in C.

**Review by moigagoo — APPROVED** [review]
I'm still not quite happy with some details, but it'll be faster for me to just fix them myself after merge.

**Pull request merged**